app/lib/analyzer/pana_runner.dart

```diff
@@ -136,17 +136,28 @@ class AnalyzerJobProcessor extends JobProcessor {
     } else {
       summary = applyPlatformOverride(summary);
       summary = await _expandSummary(summary, packageStatus.age);
-      final bool lastRunWithErrors =
-          summary.suggestions?.where((s) => s.isError)?.isNotEmpty ?? false;
+      final analyzeProcessFailed =
+          summary.health?.analyzeProcessFailed ?? false;
+      final lastRunWithErrors = analyzeProcessFailed ||
+          (summary.suggestions?.where((s) => s.isError)?.isNotEmpty ?? false);
+      final bool isLegacy = lastRunWithErrors &&
```

Comment:
I don't understand this. How can the package both have analysis errors and have failed to resolve dependencies? I thought failing to pub get means analysis doesn't happen?

I fear this is too broad. Having analysis errors or dependencies failing to resolve doesn't mean it's a Dart 1 package, unless it's a SDK version constraint bug (and even there, could be a package with a higher bound than we run on pub).

Reply:
> I don't understand this. How can the package both have analysis errors and have failed to resolve dependencies?

We have several classes of errors. The ones you refer to are coming from `dartanalyzer`, and they are tracked in `Health.analyzerErrorCount`.

Here, it can be any issue coming form the tool-enviroment, e.g. `pub upgrade` failing due to network errors. In such case we have errors, but not in the category that would classify it as a legacy.

Reply:
> Having analysis errors or dependencies failing to resolve doesn't mean it's a Dart 1 package, unless it's a SDK version constraint bug (and even there, could be a package with a higher bound than we run on pub).

There is a subset of cases where this could be a legitimate dependency resolution error, that's right.